content/blog/2023/02/2023-02-27-build-msi-locally.adoc

```diff
@@ -0,0 +1,186 @@
+---
+layout: post
+title: "How to build an unsigned Jenkins MSI on your Windows machine"
+tags:
+- jenkins
+- windows
+- msi
+authors:
+- gounthar
+- notmyfault
+discourse: true
+---
+
+Should you ever need to rebuild a Jenkins MSI on your Windows machine, here is a way to do it.
+
+== Pre-requisites
+
+=== Jenkins WAR file
+
+First of all, you should get the Jenkins war file that will be inside that MSI file.
+You can get it from the official Jenkins website or from https://updates.jenkins.io/[the Jenkins update center].
+
+Check https://www.jenkins.io/download/[Jenkins download page] and download the latest weekly version of Jenkins for example.
+You could always hit https://updates.jenkins.io/latest/jenkins.war[the direct link] to get the latest weekly version, but then you wouldn't know which version number you are using. Just saying.
+
+=== Git
+
+There are quite a few ways to get Git on Windows, but the most straightforward is to see what the https://git-scm.com/download/win[official Git website recommends].
+
+=== Install MSBuild
+
+You can install https://aka.ms/vs/17/release/vs_BuildTools.exe[MSBuild] from Visual Studio or from the https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2022[Build Tools for Visual Studio].
+
+This command line tool is used to build the MSI file.
+
+=== Install .NET Framework 3.5
+
+You may already have it installed on your machine, but not activated.
+You can activate it from the Windows Features dialog box.
+
+To access this dialog box, press the keys +++<kbd>+++⊞ Win+++</kbd>+++ + +++<kbd>+++R+++</kbd>+++, then enter the command `appwiz.cpl` and push enter.
+Search for
+
+____
+Turn Windows features on or off.
+____
+
+Tick the `.NET Framework 3.5` entry and install.
+
+IMPORTANT: now run Windows Update to check for security updates.
+
+If it is not installed yet, you can install https://dotnet.microsoft.com/en-us/download/dotnet-framework/net35-sp1[.NET Framework 3.5] from the Windows Features.
+
+=== Check if you have PowerShell, install it if not
+
+In recent versions of Windows, PowerShell is already installed and accessible through the https://support.microsoft.com/en-us/topic/6453ce98-da91-476f-8651-5c14d5777c20#:~:text=In%20Windows%2011%2022H2%2C%20the,an%20instance%20of%20Windows%20Terminal[terminal] application.
+At the time of writing, the pre-installed version is `5.1.22621.963`.
+You can also install the latest version from the https://www.microsoft.com/en-us/p/powershell/9mz1snwt0n5d?activetab=pivot:overviewtab[Microsoft Store] (7.3.2 at the time of writing). +
+You could also install PowerShell from GitHub by issuing the following command:
```

Comment:
```suggestion
In recent versions of Windows, PowerShell is already installed and accessible through the link:https://support.microsoft.com/en-us/windows/command-prompt-and-windows-powershell-for-windows-11-6453ce98-da91-476f-8651-5c14d5777c20[terminal] application.
At the time of writing, the pre-installed version is `5.1.22621.963`.
You can also install the latest version from the link:https://apps.microsoft.com/store/detail/powershell/9MZ1SNWT0N5D[Microsoft Store] (7.3.2 at the time of writing). +

You could also install PowerShell from GitHub by issuing the following command:
```
small suggestion to update the links to macros, if the links are incorrect please ignore